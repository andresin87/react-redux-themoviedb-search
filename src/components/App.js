import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import styles from './App.module.css';

import configureStore from '../config/store';
import Search from './Search/Search';
import Detail from './Detail/Detail';

const store = configureStore();

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className={styles['App']}>
          <header className={styles['App-header']}>
            <h1 className={styles['App-title']}>The Movie Database API</h1>
          </header>
          <div className={styles['App-body']}>
            <Router>
              <div>
                <Route exact path="/" component={Search} />
                <Route path="/detail" component={Detail} />
              </div>
            </Router>
          </div>
        </div>
      </Provider>
    );
  }
}

export default App;
