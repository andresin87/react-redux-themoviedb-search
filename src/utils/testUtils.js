import React from 'react';

export const mockComponent = (name, props) => {
	const { children, ...restOfProps } = props;
	return (
		<mocked name={name} props={restOfProps}>
			{children}
		</mocked>
	);
};

export const flushPromises = () => {
	return new Promise(resolve => {
		setTimeout(resolve, 0);
	});
};
