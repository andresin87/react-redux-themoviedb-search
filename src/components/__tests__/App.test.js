import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mockComponent } from '../../utils/testUtils';
import { fetchPopular, fetchGenres } from '../../actions';

import App from '../App';

jest.mock('../../actions');

jest.mock('../Search/SearchBar', () => {
  return props => mockComponent('SearchBar', props);
});
jest.mock('../Search/SearchTable', () => {
  return props => mockComponent('SearchTable', props);
});

describe('App component', () => {
  const component = <App />;

  beforeEach(() => {
    fetchPopular.mockReturnValue({ type: 'fetchPopular' });
    fetchGenres.mockReturnValue({ type: 'fetchGenres' });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
