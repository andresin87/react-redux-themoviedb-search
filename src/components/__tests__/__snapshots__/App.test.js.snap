// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot matchs 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      The Movie Database API
    </h1>
  </header>
  <div
    className="App-body"
  >
    <div>
      <code
        name="SearchBar"
        props={Object {}}
      />
      <code
        name="SearchTable"
        props={Object {}}
      />
    </div>
  </div>
</div>
`;
